code/game/machinery/computer/buildandrepair.dm

```diff
@@ -113,9 +113,40 @@
 			if(P.tool_behaviour == TOOL_SCREWDRIVER)
 				P.play_tool_sound(src)
 				to_chat(user, "<span class='notice'>You connect the monitor.</span>")
-				var/obj/B = new circuit.build_path(loc)
-				B.setDir(dir)
-				transfer_fingerprints_to(B)
+
+				var/obj/machinery/computer/new_computer = new circuit.build_path(loc)
+				new_computer.setDir(dir)
+				transfer_fingerprints_to(new_computer)
+
+				if(istype(new_computer))
+					// Machines will init with a set of default components. Move to nullspace so we don't trigger handle_atom_del, then qdel.
+					// Finally, replace with this frame's parts.
+					if(new_computer.circuit)
+						// Move to nullspace and delete.
+						new_computer.circuit.moveToNullspace()
+						QDEL_NULL(new_computer.circuit)
```

Comment:
movables get sent to nullspace on `Destroy()`, is there need to do it here?

Reply:
To prevent handle_atom_del from triggering.

If any component_part in contents is qdel'd, handle_atom_del will deconstruct the machine.

Moving to nullspace prevents that from happening as this is more a special case component swap than a component deletion.

Reply:
👍

Reply:
Robustified the existing comment to explain the whys so what is going on is less cryptic in general.